framework/src/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -1359,13 +1359,25 @@ public Void visitUnary(UnaryTree node, Void p) {
                 || (node.getKind() == Tree.Kind.POSTFIX_DECREMENT)
                 || (node.getKind() == Tree.Kind.POSTFIX_INCREMENT)) {
             AnnotatedTypeMirror varType = atypeFactory.getAnnotatedTypeLhs(node.getExpression());
-            // For postfix increments/decrements, the value type is incorrect due to the workaround
-            // in GenericAnnotatedTypeFactory.addComputedTypeAnnotations(Tree, AnnotatedTypeMirror, boolean)
-            // for the following bug:
-            // See Issue 867: https://github.com/typetools/checker-framework/issues/867
-            // This means could result in a false warning (false positive) in some cases and a lack
-            // of a warning in other cases (false negative).
-            AnnotatedTypeMirror valueType = atypeFactory.getAnnotatedType(node);
+            Tree newValue = node;
+            AnnotatedTypeMirror valueType = atypeFactory.getAnnotatedType(newValue);
+
+            // When the operation is postfix, valueType is a type of a value before the
+            // assignment. We override valueType with a type from a store, if it's provided.
+            if (node.getKind() == Tree.Kind.POSTFIX_DECREMENT
+                    || node.getKind() == Tree.Kind.POSTFIX_INCREMENT) {
+                CFAbstractStore<?, ?> store = atypeFactory.getStoreAfter(newValue);
+                if (store != null) {
+                    CFAbstractValue<?> value =
```

Comment:
I'll explain this logic and the logic which I removed in `GenericATF` little bit more using example:
```
int x = 0;
@IntVal(0) int y = x++;
```

Before this PR, the inferred value of `x++` is `1` but it should be `0`. 
To accept `@IntVal(0) int y = x++;`, the workaround in `GenericATF`is required.

But in this PR, it has been fixed, so it's no longer necessary.

However when we consider following code:
```
@IntVal(0) int zero;
zero++;
```

The compound assignment by `zero++` should be error because `1` is assigned to `zero`.
But the inferred value of `zero++` and `zero` are `0`, so `getInferredValueFor(node.getExpression())` or `getInferredValueFor(node)` doesn't works well in this situation.
The value of of `zero` in the store after `zero++` is `1`.
So I've add the logic to `visitUnary`, which takes a store after the node and takes a value from the store and updates the type.
I think this logic should be in `visitUnary` because this logic is necessary just only in `visitUnary` and it is close to what `visitUnary` want to do.